features/FEATURE_LWIP/lwip-interface/LWIPMemoryManager.cpp

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "pbuf.h"
+#include "LWIPMemoryManager.h"
+
+emac_mem_buf_t *LWIPMemoryManager::alloc_heap(uint32_t size, uint32_t align)
+{
+    struct pbuf *pbuf = pbuf_alloc(PBUF_RAW, size + align, PBUF_RAM);
+    if (pbuf == NULL) {
+        return NULL;
+    }
+
+    if (align) {
+        uint32_t remainder = (uint32_t)pbuf->payload % align;
+        uint32_t offset = align - remainder;
+        if (offset >= align) {
+            offset = align;
+        }
+
+        pbuf->payload = (void*)((char*)pbuf->payload + offset);
+        pbuf->tot_len -= offset;
+        pbuf->len -= offset;
+    }
+
+    return (emac_mem_buf_t*)pbuf;
+}
+
+emac_mem_buf_t *LWIPMemoryManager::alloc_pool(uint32_t size, uint32_t align)
+{
+    struct pbuf *pbuf = pbuf_alloc(PBUF_RAW, size + align, PBUF_POOL);
+    if (pbuf == NULL) {
+        return NULL;
+    }
+
+    if (align) {
+        uint32_t remainder = (uint32_t) pbuf->payload % align;
+        uint32_t offset = align - remainder;
+        if (offset >= align) {
+            offset = align;
+        }
+
+        pbuf->payload = (void *)((char *) pbuf->payload + offset);
+        pbuf->tot_len -= offset;
+        pbuf->len -= offset;
+    }
+
+    return (emac_mem_buf_t*)pbuf;
```

Comment:
Could you make all the casting here C++ `static_cast`? Would help enforce the const-correctness I'm calling for elsewhere.